src/core/qgsmaphittest.cpp

```diff
@@ -205,3 +206,17 @@ void QgsMapHitTest::runHitTestLayer( QgsVectorLayer *vl, SymbolSet &usedSymbols,
   }
 }
 
+bool QgsMapHitTest::rasterVisible( QgsRasterLayer *layer ) const
+{
+  if ( ! layer->dataProvider() )
+    return false;
+  QgsRectangle footprint = layer->dataProvider()->extent();
+  if ( mSettings.destinationCrs() != layer->crs() )
+  {
+    QgsCoordinateTransform ct = QgsCoordinateTransform( mSettings.destinationCrs(), layer->crs(), mSettings.transformContext() );
+    footprint = ct.transformBoundingBox( footprint );
+  }
+  if ( mSettings.extent().intersects( footprint ) )
```

Comment:
I'm not too familiar with the class, but shouldn't `visibleExtent()` be used instead?

Reply:
Yes, definitely

Reply:
There is also 'vibiblePolygon()', doc states it can be rotated, unsure if this is better.